plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp

```diff
@@ -269,39 +269,55 @@ void MpcImportWindow::addObjects()
 	}
 	//qDebug() << "Checked:" << checkedObjectsNames;
 
+	// collect all new candidates that were selected by the user into `approvedForAddition`
 	QList<SsoElements> approvedForAddition;
 	for (int i = 0; i < candidatesForAddition.count(); i++)
 	{
-		QString name = candidatesForAddition.at(i).value("name").toString();
+		auto candidate = candidatesForAddition.at(i);
+		QString name = candidate.value("name").toString();
 		if (checkedObjectsNames.contains(name))
-			approvedForAddition.append(candidatesForAddition.at(i));
+			approvedForAddition.append(candidate);
 	}
 
+	//qDebug() << "Approved for addition:" << approvedForAddition;
+
+	// collect all existing candidates that were selected by the use into `approvedForUpdate`r
+	// if the user opted to overwrite, those candidates are added to `approvedForAddition` instead
 	bool overwrite = ui->radioButtonOverwrite->isChecked();
 	QList<SsoElements> approvedForUpdate;
 	for (int j = 0; j < candidatesForUpdate.count(); j++)
 	{
-		QString name = candidatesForUpdate.at(j).value("name").toString();
+		auto candidate = candidatesForUpdate.at(j);
+		QString name = candidate.value("name").toString();
 		if (checkedObjectsNames.contains(name))
 		{
+			// XXX: odd... if "overwrite" is false, data is overwritten anyway.
```

Comment:
Please see history of changes to investigate the correct behavior of this code chunk

Reply:
I'm looking at 178045611b16efe1d04a8bae6d3a7e46ab2f1b6d

Reply:
Just a short note, sorry I will not have time until next week if my full attention is required.

The "overwrite" means "replace the complete entry" (handled by ssoManager->appendToSolarSystemConfigurationFile(approvedForAddition);)

updating OTOH only touches the elements configured with the updateFlags.